Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -62,7 +62,10 @@ fileprivate extension Compiler.ByteCodeGen {
       try emitCharacter(c)
 
     case let .scalar(s):
-      try emitScalar(s)
+      // A scalar always matches the same as a single scalar character. This
+      // means it must match a whole grapheme in grapheme semantic mode, but
+      // can match a single scalar in scalar semantic mode.
+      try emitCharacter(Character(s))
```

Comment:
Shouldn't this be conditional on whether we're in grapheme semantics or scalar semantics?

Reply:
`emitCharacter` will call `emitConsumeScalar` for scalar mode